lib/ts/MemArena.h

```diff
@@ -117,6 +118,21 @@ class MemArena
    */
   MemSpan alloc(size_t n);
 
+  /** Allocate and initialize a block of memory.
+
+      The template type specifies the type to create and any arguments are forwarded to the constructor. Example:
+      @code
+      struct Thing { ... };
+      Thing* thing = arena.make<Thing>(...constructor args...);
+      @endcode
+
+      Do @b not call @c delete an object created this way - that will attempt to free the memory and break. A
+      destructor may be invoked explicitly but the point of this class is that no object in it needs to be
+      deleted, the memory will all be reclaimed when the Arena is destroyed. In general it is a bad idea
+      to make objects in the Arena that own memory that is not also in the Arena.
```

Comment:
Is it possible to have `delete` calls on one of these objects fail explicitly and loudly—even if it is at run-time and not compile time?

Reply:
My experience is that it does fail quite immediately, but otherwise there's no good way to do that.

Reply:
OK.